js/private/js_binary.sh.tpl

```diff
@@ -0,0 +1,275 @@
+#!{bash}
```

Comment:
@gregmagolan can we get shellcheck running on this file? will probably catch bugs

Reply:
`js_binary.tpl.sh`?

Reply:
will do in a later PR, since this file isn't actually shell